src/params/stateParams.ts

```diff
@@ -28,7 +28,7 @@ export class StateParams {
       if (!parentParams.length) continue;
 
       for (const j in parentParams) {
-        if (inheritList.indexOf(parentParams[j]) >= 0) continue;
+        if (parents[i].params[parentParams[j]].inherit == false || inheritList.indexOf(parentParams[j]) >= 0) continue;
```

Comment:
Could you define before line 27 `parentParams = parents[i].params`  and rename parentParams to `parentParamsKeys = Object.keys(parentParams)` please?

Reply:
Good point :) https://github.com/ui-router/core/pull/834/commits/d3d557437135a60a48cc3af2015e6076fd6fef4c